### DISCUSSION TITLE
If I use JIT the mix watcher goes crazy

### DISCUSSION THREAD

**Discussion by SergiArias** [tighten/jigsaw#587] [Q&A] [answered]
JIT works, but then, mix starts to compile and recompile obsessively when doing npm run watch, and not waiting on save. Normal mode works okey. Is there something wrong in the config? I wasn't able to find the problem.

**Comment by bakerkretzmar** [comment]
Are you using the plain starter site or one of the templates? Can you share the npm script you're running, your `webpack.mix.js`, and your Tailwind `purge`/`content` config?

**Reply by SergiArias** [reply]
I am using the plain starter site, without any change to webpack.mix.js. I just used mode: 'jit' in tailwind.config.js before purge.

**Reply by bakerkretzmar** [reply]
Okay thanks, what's your purge config though? Do you have a `purge` or `content` key in your Tailwind config file?

**Reply by SergiArias** [reply]
I don't, I just sticked with the out of the box config except the jit mode:
```js
  purge: [
    'source/**/*.blade.php',
    'source/**/*.md',
    'source/**/*.html',
  ],
```

**Comment by keithbrink** [comment, marked as answer]
I recently encountered this issue and had to add `TAILWIND_MODE=watch` to my watch script:

`"watch": "TAILWIND_MODE=watch mix watch"`

**Reply by SergiArias** [reply]
I can confirm this works. It's weird because I also use roots/sage which is very similar to jigsaw but with wordpress and it doesn't have this TAILWIND MODE in its watch.

**Reply by ju5t** [reply]
In Tailwind 3 this doesn't work anymore. I am having the same issue as the original post.

**Reply by bakerkretzmar** [reply]
Yeah this is an issue with the default watcher setup, I'll dig in a bit more tomorrow. For now the fix is to only watch the actual specific **file** paths that you need—basically, don't have any `/**/` patterns in any of your watched paths. You may have to override the entire default watch config to get this to work, which I'll fix tomorrow.

**Reply by ju5t** [reply]
Is the watches also intended to rebuild Jigsaw itself or just the assets? Because I am now manually rebuilding Jigsaw all the time :)

**Reply by bakerkretzmar** [reply]
The Laravel Mix plugin should also be rebuilding the site. What version of `laravel-mix-jigsaw` are you using?

**Reply by ju5t** [reply]
I am using the latest version.

**Reply by inubayuaji** [reply]
Hey I got same issue from original post. I use tailwind 3.

**Reply by medilies** [reply]
@inubayuaji  Check this readme I made for myself https://github.com/elaborate-code/elaborate-code.github.io#using-tailwind-3 it may help. And I may change its content in the futur

**Reply by inubayuaji** [reply]
Hey @medilies thanks for the kind reply
Now I managed to find the solution, I change from this code
```
  purge: [
    'source/**/*.blade.php',
    'source/**/*.md',
    'source/**/*.html',
  ],
```
to this code
```
content: [
    "source/_components/**/*.blade.php",
    "source/_layouts/**/*.blade.php",
    "source/_posts/**/*.md",
    "source/_posts/**/*.blade.php",
    "source/paket/**/*.blade.php",
    "source/blog.blade.php",
    "source/index.blade.php",
  ],
```
What I did was omit the path to the assets folder. I noticed that when the contents of the assets changed it would call the build process again, resulting in an endless loop.
